pkg/upgrade/update.go

```diff
@@ -57,74 +58,48 @@ func (u *simpleUpgrader) Update(ctx context.Context, cs, oldCs *api.OpenShiftMan
 		return err
 	}
 
-	if isScaleOp {
-		for _, agent := range cs.Properties.AgentPoolProfiles {
-			vms, err := listVMs(ctx, cs, vmc, agent.Role)
-			if err != nil {
-				return err
-			}
+	for _, agent := range cs.Properties.AgentPoolProfiles {
+		vms, err := listVMs(ctx, cs, vmc, agent.Role)
+		if err != nil {
+			return err
+		}
 
-			// wait for newly created VMs to reach readiness
-			for _, vm := range vms {
-				hostname := *vm.VirtualMachineScaleSetVMProperties.OsProfile.ComputerName
-				if _, found := vmsBefore[hostname]; !found {
-					log.Infof("waiting for %s to be ready", hostname)
-					err = u.WaitForReady(ctx, cs, agent.Role, hostname)
-					if err != nil {
-						return err
-					}
+		// wait for newly created VMs to reach readiness
+		for _, vm := range vms {
+			hostname := *vm.VirtualMachineScaleSetVMProperties.OsProfile.ComputerName
+			if _, found := vmsBefore[hostname]; !found {
+				log.Infof("waiting for %s to be ready", hostname)
+				err = u.WaitForReady(ctx, cs, agent.Role, hostname)
+				if err != nil {
+					return err
 				}
 			}
 		}
-
-		return nil
-	}
-
-	err = u.updateInPlace(ctx, cs, ssc, vmc, api.AgentPoolProfileRoleMaster)
-	if err != nil {
-		return err
-	}
-
-	// TODO: updatePlusOne isn't good enough to avoid interruption on our infra
-	// nodes.
-	err = u.updatePlusOne(ctx, cs, ssc, vmc, api.AgentPoolProfileRoleInfra)
-	if err != nil {
-		return err
 	}
 
-	err = u.updatePlusOne(ctx, cs, ssc, vmc, api.AgentPoolProfileRoleCompute)
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
+	// For PP day 1, scale is permitted but not any other sort of update.  When
+	// we enable configuration changes and/or upgrades, uncomment this code.  At
+	// the same time, current thinking is that we will add a hash-based
+	// mechanism to avoid unnecessary VM rotations as well.
 
-// isScaleUpdate returns whether the update is a scale operation.
-func isScaleUpdate(cs, oldCs *api.OpenShiftManagedCluster) bool {
-	newCounts := make(map[string]int)
-	for _, new := range cs.Properties.AgentPoolProfiles {
-		newCounts[new.Name] = new.Count
-	}
+	if os.Getenv("DEVELOPMENT_ALWAYS_ROTATE") != "" {
```

Comment:
Instead of adding one level of indentation, you can exit early.
```go
if os.Getenv("DEVELOPMENT_ALWAYS_ROTATE") == "" {
   return nil
}
// update code
```

Reply:
It's possible, but this is also a temporary hack and this highlights it.